staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/helpers.go

```diff
@@ -116,3 +116,22 @@ func CRDRemoveFinalizer(crd *CustomResourceDefinition, needle string) {
 	}
 	crd.Finalizers = newFinalizers
 }
+
+func HasCRDVersion(crd *CustomResourceDefinition, version string) bool {
+	for _, v := range crd.Spec.Versions {
+		if v.Name == version {
+			return true
+		}
+	}
+	return false
+}
+
+func GetCRDStorageVersion(crd *CustomResourceDefinition) string {
```

Comment:
Godoc

Reply:
done

Reply:
Shouldn't we assume CRD is passed validation?

Reply:
Added the error handling